topdown/eval.go

```diff
@@ -2642,7 +2644,10 @@ func (e evalVirtualComplete) evalValueRule(iter unifyIterator, rule *ast.Rule, p
 
 		err := e.evalTerm(iter, term, termbindings)
 		if err != nil {
-			return err
+			ok := errors.Is(err, errEarlyExit)
+			if !ok || e.ir.EarlyExit {
```

Comment:
For my own understanding... this ensures that topdown does not (incorrectly) early-exit on "q" when the early-exit signal from the closure of "p" is received here. In other words, the early-exit signal from "p" is caught here and dropped.

Reply:
Again and again I'm getting unsure about this spoiling EE for valid cases, in more involved situations... My gut feeling is that "the last" eval of q should pass the EE error along. I'll have another look into this, maybe add another trace testcase.